superset-frontend/plugins/plugin-chart-echarts/src/Timeseries/transformProps.ts

```diff
@@ -149,8 +148,9 @@ export default function transformProps(
 
   const colorScale = CategoricalColorNamespace.getScale(colorScheme as string);
   const rebasedData = rebaseForecastDatum(data, verboseMap);
+  // todo: if the both granularity_sqla and x_axis are `null`, should throw an error
   const xAxisCol =
-    verboseMap[xAxisOrig] || getColumnLabel(xAxisOrig || DTTM_ALIAS);
+    verboseMap[xAxisOrig] || getAxis(chartProps.rawFormData) || DTTM_ALIAS;
```

Comment:
If we make `getAxis` always return a non-undefined value, then we won't need to have the ` || DTTM_ALIAS` here, right? It feels slightly redundant, as it feels like the `getAxis` helper should always return a value (or raise if it can't).